pkg/bpmn_engine/engine.go

```diff
@@ -44,17 +45,14 @@ func New(name string) BpmnEngineState {
 // CreateInstance creates a new instance for a process with given processKey
 // will return (nil, nil), when no process with given was found
 func (state *BpmnEngineState) CreateInstance(processKey int64, variableContext map[string]interface{}) (*ProcessInstanceInfo, error) {
-	if variableContext == nil {
-		variableContext = map[string]interface{}{}
-	}
 	for _, process := range state.processes {
 		if process.ProcessKey == processKey {
 			processInstanceInfo := ProcessInstanceInfo{
-				processInfo:     &process,
-				instanceKey:     state.generateKey(),
-				variableContext: variableContext,
-				createdAt:       time.Now(),
-				state:           process_instance.READY,
+				processInfo: &process,
+				instanceKey: state.generateKey(),
+				scope:       variable_scope.NewScope(nil, variableContext),
```

Comment:
the field should still be named "variableContext". please rename it back.
the concept of scopes, only works with multiple variableContexts.

Reply:
I have renamed it back, thantks very much.